lambda_functions/two_way_sms/two_way_sms_v2.py

```diff
@@ -0,0 +1,431 @@
+        return False
+
+    return True
+
+def valid_event(event_data: dict) -> bool:
+    """
+    Ensure the event has all the necessary fields
+    """
+    try:
+        if 'Records' in event_data:
+            for record in event_data.get('Records'):
+                if record.get('body') is None:
+                    # Log specific record causing issues
+                    logger.critical(f'Failed to retrieve body in record: {record}')
+                    return False
+        return True
+    except Exception as e:
+        logger.critical(f'Failed to parse event_data')
+        return False
+
+def forward_to_service(inbound_sms: dict, url: str) -> bool:
+    """
+    Forwards inbound SMS to the service that has 2-way SMS setup.
+    """
+    try:
+        if url is None:
+            logger.error("No URL provided in configuration for service")
+            return False
+
+        logger.debug(f'Connecting to {url}, sending: {inbound_sms}')
+
+        headers = {
+            'Content-type': 'application/json'
+        }
+
+        response = requests.post(
+            url,            
+            verify=False,
+            json=inbound_sms,
+            timeout=TIMEOUT,
+            headers=headers
+        )
+
+        logger.info(f'Response Status: {response.status_code}')
+        logger.debug(f"Response Content: {response.content}")
+
+        return True
+    except requests.HTTPError as e:
+        logger.error("HTTPError With Http Request")
+        logger.exception(e)
+    except requests.RequestException as e:
+        logger.error("RequestException With Http Request")
+        logger.exception(e)
+    except Exception as e:
+        logger.error("General Exception With Http Request")
+        logger.exception(e)
+        # Need to raise here to move the message to the dead letter queue instead of retry. 
+        raise
+
+    return False
+
+def push_to_sqs(inbound_sms: dict, is_retry: bool) -> None:
+    """
+    Pushes an inbound sms or entire event to SQS. Sends to RETRY or DEAD LETTER queue dependent
+    on is_retry variable. 
+    """
+    # NOTE: https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/API_SendMessage.html
+    # Need to ensure none of those unicode characters are in the message or it's gone
+    try:
+        logger.warning(f'Pushing event to {"RETRY" if is_retry else "DEAD LETTER"} queue')
+        logger.debug(f'Event: {inbound_sms}')
+
+        queue_msg = json.dumps(inbound_sms)
+
+        aws_sqs_client.send_message(QueueUrl=RETRY_SQS_URL if is_retry else DEAD_LETTER_SQS_URL,
+                                    MessageBody=queue_msg)
+
+        logger.info('Completed enqueue of message')
+    except Exception as e:
+        logger.exception(e)
+        logger.critical(f'Failed to push event to SQS: {inbound_sms}')
+        if is_retry:
+            push_to_sqs(inbound_sms, False)
+        else:
+            logger.critical(f'Attempt to enqueue to DEAD LETTER failed')
```

Comment:
```suggestion
            logger.critical('Attempt to enqueue to DEAD LETTER failed')
```
No need for f-string